### PR TITLE
Odin Project: Associations

### PR DESCRIPTION
**Pull request by kwizl**
# Private Events Project

- This project entails making a web application with the basic functionality of the Eventbrite web application. A user registers, then they have the authorization to login to the web app. They can create events an also view the events created by other users.

### CODE REVIEW

**Review by Ispirett — CHANGES_REQUESTED** [review]
## Status Changes Required
Firstly Great job so far on this project you doing well @kwizl  @abdelp. :+1: 
There are just a few issues to take care of.

- [x] There should be a link to take you too the events show page.
- [x] I can't seem to get myself added to an event.
- [x] Upcoming events and past events do not seem to be working correctly. When I add events it never shows up in either.

**Comment by abdelp** [comment]
the upcoming and past events show up when a user pushes the "Assist" button

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required
Great work so far team :clap: 

**REQUIRED**
* There is an error when I press assist button. Check this out :point_down: 
Screenshot from 2020-05-09 00-51-22
* Helper methods for the controllers should be placed in the `ApplicationController`. Only views related logic should reside in `app/helpers`.  For example `assist_event` method in `app/helpers` coud be moved to the application controller. **Resources**:
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

**After implementing the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class EventsController < ApplicationController
+  helper_method :current_user, :assist_event
+  before_action :user_authenticated
+  include EventsHelper
+
+  def index
+    @events = Event.select('events.*, users.id user_id, users.name creator_name').joins(:creator)
```
Comment (uryelah) [inline app/controllers/events_controller.rb:7]: The EventsController#index:

- [x] lists past events, i.e. @past_events = Event.previous_events

- [x] lists ucomping events, i.e. @upcoming_events = Event.upcoming_events

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      if @user.save
+        session[:name] = params[:name]
+        format.html { redirect_to users_path, notice: 'User was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def show
```
Comment (uryelah) [inline app/controllers/users_controller.rb:26]: The UsersController#show:

- [x] lists events created by user, i.e. @created_events =  current_user.events

File: app/helpers/events_helper.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module EventsHelper
+  def event_params
```
Comment (uryelah) [inline app/helpers/events_helper.rb]: - [x] If a method is only needed in the controller it should be a private method in its controller.
The methods in the helpers are the ones you need to access in the views and should not connect directly to model methods. EX(Post.new, Comment.create, current_user.friends.find_by(...)...)

- Check the link in the previous review about where you should put your code in rails to keep the project tidy.

File: app/helpers/user_helper.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module UserHelper
+  def user_params
```
Comment (uryelah) [inline app/helpers/user_helper.rb]: - [x] This method should be a private method inside UsersController

File: app/views/events/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <tr>
+      <th scope="col">Name</th>
+      <th scope="col">Description</th>
+      <th scope="col">Date</th>
+      <th scope="col">Creator</th>
+      <th scope="col"></th>
+    </th>
+  </thead>
+  <tbody>
+    <% @events.each do |e| %>
```
Comment (uryelah) [inline app/views/events/index.html.erb]: - [ ] There should be no logic at all( if else, each, etc) in the views.
View logic should be inside the pertinent helper.

File: app/views/events/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    <% if !@current_user.attended_events.exists?(@event.id) %>
+      <%= button_to "Assist", assist_event_path(event_id: @event.id), {class: 'btn btn-primary'} %>
+    <% else %>
+      <%= button_to "Assist", assist_event_path(event_id: @event.id), {class: 'btn btn-secondary', disabled: true} %>
+    <% end %>
+  </div>
+
+  <h2 class="text-center text-primary m-4">Attendees</h2>
+
+  <% @attendees.each do |a| %>
```
Comment (uryelah) [inline app/views/events/show.html.erb]: - [ ] There should be no logic at all( if else, each, etc) in the views.
View logic should be inside the pertinent helper.

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    <tr>
+        <th class="table-secondary">
+            Name
+        </th>
+        <th class="table-secondary">Description</th>
+        <th class="table-secondary">Date</th>
+        <th class="table-secondary">Creator</th>
+        <th class="table-secondary"></th>
+    </tr>
+    <% @upcoming_events.each do |e| %>
```
Comment (uryelah) [inline app/views/users/show.html.erb]: - [ ] There should be no logic at all( if else, each, etc) in the views.
View logic should be inside the pertinent helper.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hello **Abdel Pérez Téllez & Martin Njoroge**, I'm Sarah, your reviewer for this PR 👩‍🚀  

Good job with this project! The interface looks nice and it seems like all the functionality is mostly there. 

There are still some issues to fix with your code organization and small requirements from Odin project, [make sure to read the link and comment from the last TSE reviewer](http://codefol.io/posts/Where-Do-I-Put-My-Code/), check the comments below 🦾 

--------

## Changes Required 🏗️ 🐛 💡

**Comment by uryelah** [comment]
**Kindly fix the pointed out issues before requesting the next code review**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by uryelah** [comment]
Good job with most of the changes guys!

You still need to take the logic out of the views and into the helpers though.

[Maybe this StackOverflow thread can give you some direction.](https://stackoverflow.com/questions/53130326/rails-return-html-from-helper-method)

--------

## Changes Required 🏗️ 🐛 💡

- [x] Remove any logic from the views.

**Review by mosesogwo — APPROVED** [review]
**Great! Project is approved.**

_**Good luck in your next project**_

### Status: Approved :heavy_check_mark:

**Pull request merged**